src/ui/code/settingsPage.ts

```diff
@@ -2186,6 +2219,10 @@ export class settingsPage {
 			settingsPage.updateProfileGridsLayout(pageSmartProfile);
 			settingsPage.selectAddNewProfileMenu();
 
+			settingsPage.unsavedProfile = pageSmartProfile;
```

Comment:
I'm not happy to keep a reference of `pageSmartProfile` in the page level, we can be more smart about this.

Reply:
Yes, i don't like this too, but i couldn't  find a way to find if there's  an unsaved profile (and get its page object)  when leaving the page.